contracts/staking/ValidatorManager.sol

```diff
@@ -269,49 +269,43 @@ abstract contract ValidatorManager is
 
         // Get the Warp message.
         (WarpMessage memory warpMessage, bool valid) =
             WARP_MESSENGER.getVerifiedWarpMessage(messageIndex);
         require(valid, "ValidatorManager: Invalid warp message");
+        require(
+            warpMessage.sourceChainID == $._pChainBlockchainID,
+            "ValidatorManager: Invalid source chain ID"
+        );
+        require(
+            warpMessage.originSenderAddress == address(0),
+            "ValidatorManager: Invalid origin sender address"
+        );
 
-        bytes32 validationID;
-        if (setWeightMessageType) {
-            uint64 weight;
-            (validationID,, weight) =
-                ValidatorMessages.unpackSetSubnetValidatorWeightMessage(warpMessage.payload);
-            require(weight == 0, "ValidatorManager: Weight must be 0");
-        } else {
-            bool validRegistration;
-            (validationID, validRegistration) =
-                ValidatorMessages.unpackSubnetValidatorRegistrationMessage(warpMessage.payload);
-            require(!validRegistration, "ValidatorManager: Registration still valid");
-        }
+        (bytes32 validationID, bool validRegistration) =
+            ValidatorMessages.unpackSubnetValidatorRegistrationMessage(warpMessage.payload);
+        require(!validRegistration, "ValidatorManager: Registration still valid");
 
-        // Note: The validation status is not necessarily PendingRemoved here. It could be PendingAdded if the
-        // registration was never successfully registered on the P-Chain, or it could be Active if the validator
-        // removed themselves directly on the P-Chain without going through the contract.
+        // The validation status is PendingRemoved if validator removal was initiated with a call to initiateEndValidation.
+        // The validation status is PendingAdded if the validator was never registered on the P-Chain.
+        // All other statuses are unexpected, but we defensively allow them to specify the validation period as invalidated,
+        // as long as a valid SubnetValidatorRegistration message from the P-Chain is received.
         Validator memory validator = $._validationPeriods[validationID];
-        require(_msgSender() == validator.owner, "ValidatorManager: Sender not validator owner");
-
         ValidatorStatus endStatus;
-        if (
-            validator.status == ValidatorStatus.PendingRemoved
-                || validator.status == ValidatorStatus.Active
-        ) {
-            // Remove the validator from the active validators mapping.
-            delete $._activeValidators[validator.nodeID];
+        if (validator.status == ValidatorStatus.PendingRemoved) {
             endStatus = ValidatorStatus.Completed;
         } else {
             endStatus = ValidatorStatus.Invalidated;
         }
+        // Remove the validator from the active validators mapping.
+        delete $._activeValidators[validator.nodeID];
```

Comment:
we have pending registered messages saved, do we need pending end messages too?

Reply:
Added in https://github.com/ava-labs/teleporter/pull/504